### ISSUE TITLE
Node JS SDK: Generated code does not pass linter

### ISSUE THREAD

**Issue by aluzzardi** [dagger/dagger#3862]
`sdk/nodejs/.eslintrc.cjs` explicitly ignores `api/client.gen.ts`. Both Python and Go produces valid linted code, NodeJS should do the same.

/cc @slumbering @TomChv @dolanor 

**EDIT**: Replaced `api/types.ts` by `api/client.gen.ts`, issue remains the same

**Comment by slumbering** [comment]
> `sdk/nodejs/.eslintrc.cjs` explicitly ignores `api/types.ts`. Both Python and Go produces valid linted code, NodeJS should do the same.
> 
> /cc @slumbering @TomChv @dolanor

`types.ts` file is not used anymore. It was usefull before codegen PR that has been merged yesterday. It's an oversight. Thanks

**Comment by gerhard** [comment]
That's a good point. Why are the generated files not linted @slumbering @dolanor?

```diff
diff --git a/sdk/nodejs/.eslintrc.cjs b/sdk/nodejs/.eslintrc.cjs
index 3e85bf3c..6ab75b87 100644
--- a/sdk/nodejs/.eslintrc.cjs
+++ b/sdk/nodejs/.eslintrc.cjs
@@ -9,7 +9,7 @@ module.exports = {
     "dist/",

     // FIXME: generated files should be linted
-    "api/client.gen.ts",
+    // "api/client.gen.ts",
   ],
   parser: "@typescript-eslint/parser",
   plugins: ["@typescript-eslint"],
```

And then running `./hack/make sdk:nodejs:lint` results in:

<img width="1367" alt="image" src="https://user-images.githubusercontent.com/3342/202221508-482a2025-2907-46c2-9744-f6344291aad1.png">

**Comment by aluzzardi** [comment]
I've created some confusion, let me recap:

- `api/types.ts` was ignored by the linter
- I assumed we were ignoring all generated code
  -  but turns out, `api/client.gen.ts` is the generated code. `types.ts` was an old generated file
  - so we were not ignoring the generated code
- Problem 1: linter config was ignoring warnings, just printing them not failing
  - I changed that and saw there were a few in the generated code
  - I'm the one who added `client.gen.ts` to the ignore pattern, thinking of completing `types.ts`
- Problem 2: prettier was not enabled (consistent spaces/tabs/quotes/etc)
  - This created a bunch of NEW errors
  - But before yesterday, the generated code only had a few problems

**Comment by vikram-dagger** [comment]
@slumbering cc @aluzzardi : we are assuming in https://github.com/dagger/dagger/pull/3832#issuecomment-1317644433 that the auto-generated client has been linted and type checked. The plugin configuration used in that PR is partly based on this assumption. Is that assumption correct?

**Comment by aluzzardi** [comment]
> @slumbering cc @aluzzardi : we are assuming in https://github.com/dagger/dagger/pull/3832#issuecomment-1317644433 that the auto-generated client has been linted and type checked. The plugin configuration used in that PR is partly based on this assumption. Is this assumption correct?

auto-generated code is not linted, this issue is still valid

**Comment by slumbering** [comment]
@aluzzardi . I've just created a PR that lint the auto-generated file + fix some typescript issues.
👉 #3895

**Comment by aluzzardi** [comment]
Great!

@dolanor @slumbering We also need to update codegen to output linted code, otherwise next time we re-generate the API this will break gain. There's a mix of indentation, single vs double quotes, semi-colon.

**Comment by slumbering** [comment]
Related discussion on #3912 

The Python SDK uses Python for codegen which make it easy to lint while generating. AFAIK we need to use eslint to stay consistent in code formating. Generated and classic files will be rules by one entity, in our case eslint.
What are your thoughts @dolanor

**Comment by helderco** [comment]
To be fair, the Python advantage is only that it has an executable that can be used in `pyproject.toml` so we can generate and call the formatter. It's not that it's written in Python necessarily.

```toml
[tool.poe.tasks.generate]
sequence = [
    "dagger-py generate --output ${GEN_PATH}/gen.py",
    "dagger-py generate --output ${GEN_PATH}/gen_sync.py --sync",
    "isort ${GEN_PATH}/gen*.py",
    "black ${GEN_PATH}/gen*.py",
]
```

You can do the same in `mage sdk:nodejs:generate`, calling a `yarn fmt` after it's done generating.

**Comment by helderco** [comment]
I would prefer to produce already linted code but that was faster and simpler.

**Comment by slumbering** [comment]
Let's discuss about this topic today @dolanor

**Comment by slumbering** [comment]
@dolanor working on formating and linting the code generated with eslint in the mage file.

**Comment by dolanor** [comment]
fixed in d5e29ad8b672011eecbddf3d4091847d1a5d7ddf #3755
